### ISSUE TITLE
Can't create port forwarding rule

### ISSUE THREAD

**Issue by synergiator** [apache/cloudstack-terraform-provider#21]
With this code fragment,

```
resource "cloudstack_port_forward" "default" {
  ip_address_id = cloudstack_ipaddress.my_ip.id

  forward {
    protocol           = "tcp"
    private_port       = 22
    public_port        = 22
    virtual_machine_id = cloudstack_instance.my_vm.id
  }
}
```

I constantly get this error (API 4.11)

```
module.vpc01.cloudstack_network.private[0]: Refreshing state... [id=fbc1a148-d2e3-4b35-ae1c-1ed8c4973829]
module.vpc01.cloudstack_ipaddress.my_ip: Refreshing state... [id=65231fc6-97a4-4370-9f65-59b38de81cdb]
module.vpc01.cloudstack_instance.my_vm: Refreshing state... [id=1d25e677-47d6-484e-a3c9-c1a54b5cfe42]
╷
│ Error: insufficient items for attribute "forward"; must have at least 1
│ 
│   with module.vpc01.cloudstack_port_forward.default,
│   on ../modules/core/customvpc/main.tf line 55, in resource "cloudstack_port_forward" "default":
│   55: resource "cloudstack_port_forward" "default" {
│ 
```

Which item is meant to be missing here - the VM ID? Could it be some synchronization problem? - but, the IDs of the VM and the IP are actually there.

**Comment by harikrishna-patnala** [comment]
@synergiator on the first try of creating port forwarding rule, I did not find any issue it got successfully created.

Then I tried few more scenarios to purposefully make it fail, so I have applied static nat on the IP and applied PF again then it failed with proper error "loudStack API error 431 (CSExceptionErrorCode: 4350): Unable to create port forwarding rule; ip id=6 has static nat enabled"

Upon retry of apply PF rule I've faced the issue that you reported which is "Error: insufficient items for attribute "forward"; must have at least 1". Then I found the issue.

The issue issue is, upon PF rule creation failure, provider is adding the details to terraform state. So upon retry it is not finding any difference since terraform state already has it. I'll work on this to fix the issue.
